### PR TITLE
Add Inventories API endpoints

### PR DESCRIPTION
**Pull request by mauromedda**
* CreateInventory: create a new inventory
* UpdateInventory: update an inventory
* DeleteInventory:  delete an inventory
* GetInventory:    retrives inventory information using
                   the inventory id or name

The commit includes tests and usage examples.

### CODE REVIEW

File: examples/inventories.md [hunk 1]
```diff
@@ -24,3 +24,93 @@ fun main() {
...
+    if err != nil {
+        log.Fatalf("Delete Inventories err: %s", err)
+    }
+
+    log.Println("Inventroy 5 Deleted")
+
+}
+```
+
+> GetInventoryById
```
Comment (Colstuwjx) [inline examples/inventories.md]: Rename to `GetInventory` avoid confusing.

File: examples/inventories.md [hunk 2]
```diff
@@ -24,3 +24,93 @@ fun main() {
...
+       "kind":         "",
+       "host_filter":  "",
+       "variables":    "",
+    }, map[string]string{})
+
+    if err != nil {
+        log.Fatalf("Create Inventories err: %s", err)
+    }
+
+    log.Println("Inventory created. Inventory ID: %s", result.Inventory.ID")
```
Comment (Colstuwjx) [inline examples/inventories.md]: Should be %d, not %s

File: examples/inventories.md (hunk 1 above)
Comment (mauromedda) [reply examples/inventories.md]: fixed

File: examples/inventories.md (hunk 2 above)
Comment (mauromedda) [reply examples/inventories.md]: fixed

File: internal/mockserver/mockdata/inventories.go [hunk 3]
```diff
@@ -204,4 +204,204 @@ var (
...
+        "inventory_sources_with_failures": 0,
+        "insights_credential": null,
+        "pending_deletion": false
+    }`)
+
+	// MockedDeleteInventoryResponse mocked `Delete` endpoint response
+	MockedDeleteInventoryResponse = []byte(`{}`)
+
+	// MockedGetInventoryByIDResponse mocked `GetInventoryByID` endpoint response
+	MockedGetInventoryByIDResponse = []byte(`
```
Comment (Colstuwjx) [inline internal/mockserver/mockdata/inventories.go]: In order to keep same code style, pls rename `MockedGetInventoryByIDResponse` to `MockedGetInventoryResponse` also the comment part.

File: inventories_test.go [hunk 4]
```diff
@@ -276,3 +276,268 @@ func TestCreateInventory(t *testing.T) {
...
+	}
+
+	if !reflect.DeepEqual(result, expectUpdateInventoryResponse) {
+		log.Fatalf("UpdateInventory resp not as expected, expected: %v, resp result: %v", expectUpdateInventoryResponse, result)
+	}
+
+	log.Println("UpdateInventory passed!")
+}
+
+func TestGetInventoryByID(t *testing.T) {
```
Comment (Colstuwjx) [inline inventories_test.go]: Change name to `TestGetInventory` as well.

File: internal/mockserver/mockserver.go [hunk 5]
```diff
@@ -21,14 +21,27 @@ func (s *mockServer) PingHandler(rw http.ResponseWriter, req *http.Request) {
...
+	case req.Method == "PATCH", req.Method == "PUT":
+		result := mockdata.MockedUpdateInventoryResponse
+		rw.Write(result)
+		return
+	case req.Method == "DELETE":
+		result := mockdata.MockedDeleteInventoryResponse
+		rw.Write(result)
+		return
+	case req.RequestURI == "/api/v2/inventories/1/":
+		result := mockdata.MockedGetInventoryByIDResponse
```
Comment (Colstuwjx) [inline internal/mockserver/mockserver.go]: Change the `MockedGetInventoryByIDResponse`  to `MockedGetInventoryResponse` as well

**Comment by Colstuwjx** [comment]
Could you add a `TestDeleteInventory` case for test coverage?

File: inventories.go [hunk 6]
```diff
@@ -52,3 +55,57 @@ func (i *InventoriesService) CreateInventory(data map[string]interface{}, params
...
+	if err := CheckResponse(resp); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+
+}
+
+// Delete an inventory from AWX
+func (i *InventoriesService) Delete(id int) (*Inventory, error) {
```
Comment (Colstuwjx) [inline inventories.go]: I suggest change the func name to `DeleteInventory`, as we have lots of test cases, and the same fun name could broken the whole test.

File: examples/inventories.md [hunk 7]
```diff
@@ -24,3 +24,93 @@ fun main() {
...
+> Delete Inventory
+```
+import (
+    "log"
+    awxGo "github.com/Colstuwjx/awx-go"
+)
+
+func main() {
+    awx := awxGo.NewAWX("http://awx.your_server_host.com", "your_awx_username", "your_awx_passwd", nil)
+    result, err := awx.InventoriesService.Delete("5")
```
Comment (Colstuwjx) [inline examples/inventories.md]: Should be 5, rather than "5", as it is int type.

File: request.go [hunk 8]
```diff
@@ -149,10 +149,33 @@ func (r *Requester) Post(endpoint string, payload io.Reader, responseStruct inte
...
+// PatchJSON perform http patch request with json response
+func (r *Requester) PatchJSON(endpoint string, payload io.Reader, responseStruct interface{}, querystring map[string]string) (*http.Response, error) {
+	ar := NewAPIRequest("PATCH", endpoint, payload)
+	ar.SetHeader("Content-Type", "application/json")
+	ar.Suffix = ""
+	return r.Do(ar, &responseStruct, querystring)
+}
+
+// Delete performs http Delete request.
+func (r *Requester) DeleteInventory(endpoint string, responseStruct interface{}, querystring map[string]string) (*http.Response, error) {
```
Comment (Colstuwjx) [inline request.go]: Typo here, should be `Delete` in request level.

**Pull request merged**